CMakeLists.txt

```diff
@@ -1,89 +1,55 @@
-STRING (REGEX REPLACE ".*#define[ \t]+BAG_VER_REVISION[ \t]+([0-9]*).*$"
-    "\\1" OPENNS_VERS_RELEASE ${_bag_h_contents})
-
-#-----------------------------------------------------------------------------
-SET (OPENNS_PACKAGE "openns")
-SET (OPENNS_PACKAGE_NAME "OPENNS")
-SET (OPENNS_PACKAGE_VERSION "${OPENNS_VERS_MAJOR}.${OPENNS_VERS_MINOR}")
-SET (OPENNS_PACKAGE_VERSION_MAJOR "${OPENNS_VERS_MAJOR}.${OPENNS_VERS_MINOR}")
-SET (OPENNS_PACKAGE_VERSION_MINOR "${OPENNS_VERS_RELEASE}")
-SET (OPENNS_PACKAGE_STRING "${OPENNS_PACKAGE_NAME} ${OPENNS_PACKAGE_VERSION}")
-SET (OPENNS_PACKAGE_TARNAME "bag")
-SET (OPENNS_PACKAGE_URL "http://www.opennavsurf.org")
-SET (VERSION 1.6.3 OPENNS_PACKAGE_VERSION)
+# SWIG: use SWIG_MODULE_NAME property.
+if(POLICY CMP0086)
+    cmake_policy(SET CMP0086 NEW)
+endif()
 
-#-----------------------------------------------------------------------------
-# Option to Build Shared/Static libs, default is shared
-#-----------------------------------------------------------------------------
-OPTION (BUILD_SHARED_LIBS "Build Shared Libraries" ON)
-IF (BUILD_SHARED_LIBS)
-  SET (LIB_TYPE SHARED)
-ELSE (BUILD_SHARED_LIBS)
-  SET (LIB_TYPE STATIC)
-ENDIF (BUILD_SHARED_LIBS)
+# Disable addition of warning flags to the CMAKE_<LANG>_FLAGS by default for MSVC.
+if(POLICY CMP0092)
+    cmake_policy(SET CMP0092 NEW)
+endif()
 
-#-----------------------------------------------------------------------------
-# Option to Build against Shared/Static version of HDF5, default is shared
-#-----------------------------------------------------------------------------
-OPTION (BUILD_WITH_SHARED_HDF5 "Build with HDF5 as a shared library" ON)
+project(OPENNS LANGUAGES C CXX VERSION 2.0.0)
 
+# Options
+option(BAG_BUILD_SHARED_LIBS "Build Shared Libraries" ON)
+option(BAG_BUILD_SWIG "Build SWIG interface" OFF)
+option(BAG_BUILD_PYTHON "Build Python bindings" OFF)
+option(BAG_CI "The build is on the CI (Travis/Appveyor)." OFF)
 
-if (CMAKE_C_COMPILER_ID STREQUAL "GNU")
-    set (CMAKE_C_FLAGS "--std=gnu99 ${CMAKE_C_FLAGS}")
-    add_definitions ("-Wall")
-endif ()
+# Put the artifacts into a single directory for MSVC so running tests is easier.
+if(MSVC AND NOT BAG_CI)
+    if(NOT CMAKE_RUNTIME_OUTPUT_DIRECTORY)
+        set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/stage/${CMAKE_INSTALL_BINDIR})
+    endif()
 
-if (WIN32)
+    if(NOT CMAKE_LIBRARY_OUTPUT_DIRECTORY)
+        set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/stage/${CMAKE_INSTALL_LIBDIR})
+    endif()
 
-else()
-	# use, i.e. don't skip the full RPATH for the build tree
-	SET(CMAKE_SKIP_BUILD_RPATH  FALSE)
+    if(NOT CMAKE_ARCHIVE_OUTPUT_DIRECTORY)
+        set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/stage/${CMAKE_INSTALL_LIBDIR})
+    endif()
+endif()
 
-	# when building, don't use the install RPATH already
-	# (but later on when installing)
-	SET(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE) 
+if(CODE_COVERAGE)
```

Comment:
I am puzzled by this condition. The option CODE_COVERAGE is actually defined in the following included code-coverage.cmake.

Reply:
Resolved: https://github.com/OpenNavigationSurface/BAG/pull/57/commits/16f2b19acd47b03082eae0942c91e31ec9bfc9e2